typescript/packages/plugins/dexscreener/src/dexscreener.service.ts

```diff
@@ -0,0 +1,99 @@
+import { Tool } from "@goat-sdk/core";
+import { GetPairsByChainAndPairParameters, GetTokenPairsParameters, SearchPairsParameters } from "./parameters";
+import { DexscreenerPairResponse } from "./types";
+
+export class DexscreenerService {
+    private readonly baseUrl = "https://api.dexscreener.com/latest";
+    private readonly rateLimit = {
+        pairs: 300, // requests per minute
+        profiles: 60, // requests per minute
+    };
+
+    private lastRequestTime = 0;
+    private requestCount = 0;
+
+    private async enforceRateLimit() {
+        const now = Date.now();
+        const oneMinute = 60 * 1000;
+
+        // Reset counter if a minute has passed
+        if (now - this.lastRequestTime >= oneMinute) {
+            this.requestCount = 0;
+            this.lastRequestTime = now;
+            return;
+        }
+
+        // If we're under the limit, just increment
+        if (this.requestCount < this.rateLimit.pairs) {
+            this.requestCount++;
+            return;
+        }
+
+        // Wait until the minute is up
+        const timeToWait = oneMinute - (now - this.lastRequestTime);
+        await new Promise((resolve) => setTimeout(resolve, timeToWait));
+        this.requestCount = 1;
+        this.lastRequestTime = Date.now();
+    }
+
+    @Tool({
+        name: "dexscreener.get_pairs_by_chain_and_pair",
+        description: "Fetch pairs by chainId and pairId from Dexscreener",
+    })
+    async getPairsByChainAndPair(parameters: GetPairsByChainAndPairParameters) {
+        try {
+            await this.enforceRateLimit();
+            const url = `${this.baseUrl}/pairs/${parameters.chainId}/${parameters.pairId}`;
```

Comment:
missing the `/dex` in the url `https://api.dexscreener.com/latest/dex/pairs/{chainId}/{pairId}`